lib/compress/zstd_compress.c

```diff
@@ -2860,15 +2860,15 @@ size_t ZSTD_loadCEntropy(ZSTD_compressedBlockState_t* bs, void* workspace,
     dictPtr += 8;
 
     {   unsigned maxSymbolValue = 255;
-        unsigned hasZeroWeights;
+        unsigned hasZeroWeights = 1;
+        bs->entropy.huf.repeatMode = HUF_repeat_check;
```

Comment:
This will cause issue with `C90` compatibility, due to declaration order.
Set the value before the block.

Reply:
Hmm I would not have guessed that. Moved now

Reply:
Moved the wrong line:( just fixed